// Copyright 2024 Flower Labs GmbH. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
// ==============================================================================

syntax = "proto3";

package flwr.proto;

import "flwr/proto/transport.proto";

message Run {
  uint64 run_id = 1;
  string fab_id = 2;
  string fab_version = 3;
  map<string, Scalar> override_config = 4;
  string fab_hash = 5;
}
message GetRunRequest { uint64 run_id = 1; }
message GetRunResponse { Run run = 1; }
